File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        answer = [0] * len(temperatures)
        s = []
        
        for i in range(len(temperatures)):
            while s and temperatures[i] > temperatures[s[-1]]:
                idx = s.pop()
                answer[idx] = i - idx
            s.append(i)
        
        return answer
        

File: 0739-daily-temperatures/0739-daily-temperatures.java
import java.util.*;

class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int[] answer = new int[temperatures.length];
        Stack<Integer> s = new Stack<>();
        
        for (int i = 0; i < temperatures.length; i++) {
            while (!s.isEmpty() && temperatures[s.peek()] < temperatures[i]) {                
                int preIdx = s.pop();
                answer[preIdx] = i - preIdx;
            }
            s.push(i);
        }
        
        return answer;
        
    }
}